phc-rails/app/controllers/api/v1/event_registrations_controller.rb

```diff
@@ -46,49 +47,45 @@ def get_applied
       @services = @event_registration.services
       render json: { status: "true", services: ["Acupuncture", "Haircuts", "Massage"] }
     else
-      render json: { status: 404, message: "Did not find event registration corresponding to QR code." }
+      api_message_response(404, "Event registration with that number does not exist.")
     end
   end
 
   def update_service
     registration = EventRegistration.find_by(Number__c: params[:Number__c])
     if registration.nil?
-      render :json => { status: 404, message: "Did not find event registration corresponding to QR code." }
+      api_message_response(404, "Event registration with that number does not exist.")
       return
     end
 
     service = registration.services.find_by(name: params[:service_name])
     if service.nil?
-      render :json => { status: 404, message: "Did not find specified service for current event." }
+      api_message_response(404, "Service with that name does not exist.")
       return
     end
 
     case service.status
-    when Service.NONE
-      service.update_attribute(:status, Service.DROPIN)
-      render :json => { message: "The client is a drop-in." }
-      return
-    when Service.APPLIED
-      service.update_attribute(:status, Service.RECIEVED)
-      render :json => { message: "" }
-      return
-    when Service.RECIEVED
-      render :json => { message: "The client has recieved this service before." }
-      return
-    when Service.DROPIN
-      render :json => { message: "The client has recieved this service before." }
-      return
+    when Service.unspecified
+      service.update_attribute(:status, Service.drop_in)
+      api_message_response(200, "Client's status set to drop-in.")
+    when Service.applied
+      service.update_attribute(:status, Service.received)
+      api_message_response(200, "Client's status set to received.")
+    when Service.drop_in
+      api_message_response(200, "Client has already received service.")
+    when Service.received
+      api_message_response(200, "Client has already received service.")
     else
-      raise "Service status is not known. This should not happen!"
+      api_message_response(500)
     end
   end
 
   def update_feedback
     @event_registration = EventRegistration.find_by(Number__c: params[:Number__c])
     if @event_registration.update(event_registration_params)
-      render :json => { status: 201, message: "Good job Shimmy" }
+      api_message_response(201, "Good job Shimmy")
```

Comment:
There isn't a feedback model yet! I can make that on Monday and fully set this action up.